internal/callgraph/language/java11/stategy.go

```diff
@@ -0,0 +1,97 @@
+package java
+
+import (
+	"fmt"
+	"log"
+	"path/filepath"
+
+	"github.com/debricked/cli/internal/callgraph/cgexec"
+	conf "github.com/debricked/cli/internal/callgraph/config"
+	"github.com/debricked/cli/internal/callgraph/job"
+	"github.com/debricked/cli/internal/io/finder"
+	"github.com/debricked/cli/internal/io/writer"
+	"github.com/fatih/color"
+)
+
+type Strategy struct {
+	config conf.IConfig
+	files  []string
+	finder finder.IFinder
+	ctx    cgexec.IContext
+}
+
+func (s Strategy) Invoke() ([]job.IJob, error) {
+	var jobs []job.IJob
+	// Filter relevant files
+
+	if s.config == nil {
+		strategyWarning("No config is setup")
+
+		return jobs, nil
+	}
+
+	pmConfig := s.config.Kwargs()["pm"]
+
+	var roots []string
+	var err error
+	switch pmConfig {
+	case maven:
+		roots, err = s.finder.FindMavenRoots(s.files)
+	default:
+		roots, err = s.finder.FindMavenRoots(s.files)
+	}
+
+	if err != nil {
+		strategyWarning("Error while finding roots: " + err.Error())
+
+		return jobs, err
+	}
+
+	classDirs, _ := s.finder.FindJavaClassDirs(s.files)
+	absRoots, _ := finder.ConvertPathsToAbsPaths(roots)
+	absClassDirs, _ := finder.ConvertPathsToAbsPaths(classDirs)
+	rootClassMapping := finder.MapFilesToDir(absRoots, absClassDirs)
```

Comment:
I am a bit confused by the naming here, could we break out the "utility functionality" to a different package maybe?